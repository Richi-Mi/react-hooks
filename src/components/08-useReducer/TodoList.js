import React from 'react'
import PropTypes from 'prop-types';
import TodoListItem from './TodoListItem';

const TodoList = React.memo(({ todos, handleDelete, handleToggle }) => {
    console.log(' - Renderize TodoList')
    return (
        <ul className='list-group list-group-flush'>
            {
                todos.map(({ desc, id, done }, i) => {
                    return <TodoListItem 
                        id={ id } 
                        desc={ desc } 
                        index={ i } 
                        done={ done }
                        key={ id }
                        handleDelete={ handleDelete } 
                        handleToggle={ handleToggle } 
                    />
                })
            }
        </ul>
    )
})

TodoList.propTypes = {
    todos: PropTypes.array.isRequired,
    handleDelete: PropTypes.func.isRequired,
    handleToggle: PropTypes.func.isRequired
}

export default TodoList
