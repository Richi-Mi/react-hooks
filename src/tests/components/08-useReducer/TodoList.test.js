import { shallow } from 'enzyme';
import React from 'react';
import TodoList from '../../../components/08-useReducer/TodoList';
import demoTodos from '../../fixtures/demoTodos';

describe(' * Pruebas en <TodoList />', () => {
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow( 
        <TodoList 
            todos = { demoTodos } 
            handleDelete = { handleDelete }
            handleToggle = { handleToggle }
        /> 
    );
    test(' - Debe de Mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })
    test(' - Debe de tener 2 elementos <TodoListItem />', () => {
        expect( wrapper.find('TodoListItem').length ).toBe( demoTodos.length )    
        expect( wrapper.find('TodoListItem').at(0).prop('handleDelete') ).toEqual( expect.any( Function ) );    
    })
    
})
