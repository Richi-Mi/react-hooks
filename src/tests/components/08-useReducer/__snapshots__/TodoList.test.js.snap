// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` * Pruebas en <TodoList />  - Debe de Mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    desc="Aprender React"
    done={false}
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    id={1}
    index={0}
    key="1"
  />
  <TodoListItem
    desc="Aprender Mongo"
    done={false}
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    id={2}
    index={1}
    key="2"
  />
  <TodoListItem
    desc="Aprender Express"
    done={false}
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    id={3}
    index={2}
    key="3"
  />
</ul>
`;
